server/mdm/microsoft/wstep_test.go

```diff
@@ -0,0 +1,164 @@
+package microsoft_mdm
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"errors"
+	"math/big"
+	"strings"
+	"testing"
+
+	"github.com/micromdm/nanomdm/cryptoutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewWSTEPDepot(t *testing.T) {
+	// Test that NewWSTEPDepot returns an error if the cert PEM is invalid.
+	_, err := NewWSTEPDepot([]byte("invalid"), []byte(testKey))
+	require.Error(t, err)
+	require.ErrorContains(t, err, "failed to decode PEM certificate")
+
+	// Test that NewWSTEPDepot returns an error if the key PEM is invalid.
+	_, err = NewWSTEPDepot([]byte(testCert), []byte("invalid"))
+	require.Error(t, err)
+	require.ErrorContains(t, err, "failed to decode PEM private key")
+
+	// Test that NewWSTEPDepot returns an error if the cert PEM is not a certificate.
+	_, err = NewWSTEPDepot([]byte(testKey), []byte(testKey))
+	require.Error(t, err)
+	require.ErrorContains(t, err, "failed to decode PEM certificate")
+
+	// Test that NewWSTEPDepot returns an error if the key PEM is not a private key.
+	_, err = NewWSTEPDepot([]byte(testCert), []byte(testCert))
+	require.Error(t, err)
+	require.ErrorContains(t, err, "PEM type is not RSA PRIVATE KEY")
+
+	// Test that NewWSTEPDepot returns a *WSTEPDepot if the cert and key PEMs are valid.
+	depot, err := NewWSTEPDepot([]byte(testCert), []byte(testKey))
+	require.NoError(t, err)
+
+	// Check that the identity cert and private key were parsed correctly.
+	require.NotNil(t, depot.identityCert)
+	require.NotNil(t, depot.identityPrivateKey)
+
+	wantCert, err := cryptoutil.DecodePEMCertificate([]byte(testCert))
+	require.NoError(t, err)
+	require.Equal(t, wantCert, depot.identityCert)
+
+	wantKey, err := decodeRSAKeyFromPEM([]byte(testKey))
+	require.NoError(t, err)
+	require.Equal(t, wantKey, depot.identityPrivateKey)
+
+	// Check that the identity fingerprint was calculated correctly.
+	csum := sha1.Sum(wantCert.Raw)
+	wantFingerprint := strings.ToUpper(hex.EncodeToString(csum[:]))
+	require.Equal(t, wantFingerprint, *depot.IdentityFingerprint)
+
+	// Check that the maximum serial number was calculated correctly.
+	wantMaxSerialNumber := new(big.Int).Lsh(big.NewInt(1), 128)
+	require.Equal(t, *wantMaxSerialNumber, *depot.maxSerialNumber)
+}
+
+func TestSignClientCSR(t *testing.T) {
+	// TODO
```

Comment:
This tests should be completed when the associated function from the POC is replaced.